tests/core/pyspec/eth2spec/utils/bls.py

```diff
@@ -121,24 +171,123 @@ def AggregatePKs(pubkeys):
         # milagro_bls._AggregatePKs checks KeyValidate internally
```

Comment:
Since `py_ecc_bls` has an assert statement above and `arkworks_bls` uses `py_ecc_bls` in this function, should `arkworks_bls` have the assert statement as well?

Reply:
Ah, a complicated background story of `AggregatePKs`. `AggregatePKs` is part of the IETF standard. The CL specs only use `eth_aggregate_pubkeys` now.

To optimize the pyspec, we use this `bls.AggregatePKs` to implement `eth_aggregate_pubkeys`: https://github.com/ethereum/consensus-specs/blob/37f6abacb6c92abe3a130eb0be89768d9bcf6eff/setup.py#L67-L70

So `arkworks_bls` should do what `eth_aggregate_pubkeys` do. so yes, it should check `KeyValidate`.


But an easier solution here is, just return `milagro_bls._AggregatePKs(list(pubkeys))` because milagro does check `KeyValidate`.

Reply:
But `arkworks_bls` uses `py_ecc_bls` for this function not `milagro_bls`. So it will not do the `KeyValidate` check?

Note that I didn't intend to comment this line. I intended to comment the line of `assert all(bls.KeyValidate(pubkey) for pubkey in pubkeys)` but Github doesn't allow me to do so.